src/Components/Shared/Modal/index.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import styles from './modal.module.css';
+
+const Modal = (props) => {
+  const { children, isOpen, handleClose, isActionModal, action, actionButton } = props;
+  let buttons;
+  if (!isOpen) {
+    return null;
+  }
+  if (isActionModal) {
+    buttons = (
+      <div className={styles.buttonsList}>
+        <button className={styles.cancel} onClick={() => handleClose()}>
+          Cancel
+        </button>
+        <button
+          className={styles.confirm}
+          onClick={() => {
+            action();
+            handleClose();
+          }}
+        >
+          {actionButton}
+        </button>
+      </div>
+    );
+  } else {
+    buttons = (
+      <div className={styles.buttonsList}>
+        <button className={styles.confirm} onClick={() => handleClose()}>
+          Accept
+        </button>
+      </div>
+    );
+  }
+  return (
+    <div className={styles.modalOverlay}>
+      <div className={styles.modalWrapper}>
+        {children}
+        <div className={styles.buttonsWrapper}>{buttons}</div>
```

Comment:
I'd do something like this all inside JSX instead of having a buttons variable
```suggestion
        <div className={styles.buttonsWrapper}>
        <div className={styles.buttonsList}>
        <button className={isActionModal ? styles.cancel : styles.confirm} onClick={() => handleClose()}>
          isActionModal ? Cancel : Accept
        </button>
        isActionModal ? (
        <button
          className={styles.confirm}
          onClick={() => {
            action();
            handleClose();
          }}
        >
          {actionButton}
        </button>
        ) : null
      </div>
        </div>
```

Reply:
Another change to avoid the `null`:
```
{isActionModal && (
         <button
           className={styles.confirm}
           onClick={() => { 
            action();
            handleClose();
           }}
         >
          {actionButton}
        </button>
        )}
```